### PR TITLE
Add geocoding and display error message

### LINKED ISSUE #2: Geocoding
## Value statement 

As a **person**
I need **to just write my city or postal code**
so that **I can easy get covid-19 information about my county**

## Description (Text, Scribble, Wireframe, Design)

If the user puts the name or the postal code of his city into the search, the geocoding API gives the county back, which is used for the data fetch from the covid-19 RKI API.

capstone-project_first-version(7)

## Acceptance criteria

-   [x] Display error feedback if the county search was not successful.
-   [x] Display error feedback if the RKI API fetch was not successful.

## Tasks

-   [x] Create geocoding fetch service
-   [x] Implement fetch and create error handling
-   [x] Test on mobile and desktop

## Size

large

### PR DESCRIPTION
**Pull request by JulianBlohm**
Please review my new feature :)

The user doesn't have to type the county in. User can now enter a place or postcode and will get the incidence rate for the county the place is in. 
If the user Input doesn't fit any place in germany or there is an error with getting the corona data from the server, there will be an error displayed.

Don't mind the hook management, it's working, but I will refactor that in the next step.

### CODE REVIEW

**Review by marwinburesch — COMMENTED** [review]
lgtm!
- maybe later you can combine some of the logic, and include error and getGeoData?
🐜

File: src/MainPage.js [hunk 1]
```diff
@@ -1,43 +1,60 @@
...
     
-    const countyResult = county.charAt(0).toUpperCase() + county.slice(1)
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    useEffect(() => getCountyData(), [county])
+    useEffect(() => {userPlace && getCountyData()}, [userPlace])
+    useEffect(() => {coordinates.longitude && getIncidenceData()}, [coordinates])
+    useEffect(() => setErrorMessage(''), [userPlace,coordinates] )
+
+    console.log(coordinates)
```
Comment (JanaHaeusler) [inline src/MainPage.js]: Are ``console.log(coordinates)``and ``console.log(countyData)`` placeholders for future logics? Can be removed, if not necessary :)

File: src/components/Form.js [hunk 2]
```diff
@@ -1,36 +1,51 @@
...
+const FormStyled = styled.form`
+position: relative;
+
+span {
+    font-size: 0.8rem;
+    color: darkred;
+    position: absolute;
+    left: 11px;
+    top: 55px;
+
```
Comment (Hecktors) [inline src/components/Form.js]: The empty line is redundant

**Review by Hecktors — COMMENTED** [review]
Have nothing to complain. The only good reason for covid-19 is the creation of this app!

File: src/components/Form.js [hunk 3]
```diff
@@ -1,36 +1,51 @@
...
-                    placeholderText="Gib deinen Landkreis ein..."/>
+                    placeholderText="Gib Ort oder PLZ ein..."/>
+                <span>{errorMessage}</span>
                 <Button text="Suchen"/>
-            </form>
+            </FormStyled>
 )
 }
 
 Form.propTypes = {
```
Comment (JanaHaeusler) [inline src/components/Form.js:32]: You could think about, if props should be required.

**Review by JanaHaeusler — COMMENTED** [review]
An exciting app you are building! :) Added just a two comments. If you have any questions, feel free to get back to me. And you already said, that you want to refactor your hooks. That is great. :)

**Pull request merged**